### ISSUE TITLE
GG Max sound slots different from Classic in parameters.

### ISSUE THREAD

**Issue by AmenMoses** [TheGameCreators/GameGuruRepo#775]
GG Max advanced properties only shows 4 audio slots for an entity, i.e. slots 0-3, but GG Classic has 5 slots, i.e. slots 0-4.

**Comment by LeeBamberTGC** [comment]
Indeed. One slot (string) was given up during development to support the new LUA 'not quite as dynamic as some users are misled into thinking' System. How many slots do you think we should have. I had assumed we would be using the new DLUA system to add parameters into the script, but if you feel there is a benefit in expanding the hard-coded fields, let me know.

**Comment by AmenMoses** [comment]
I was just thinking that some users may have used all 5 slots in their maps and couldn't quite see why one had been dropped.

**Comment by AmenMoses** [comment]
Btw, some old ai scripts still use the 'sound set' mechanism but the new ai scripts seem to have dropped it, is it still supported fully?

**Comment by tarkus1971** [comment]
@Lee - I would like 10 slots if possible. Is that even doable? Regards, great work on MAX so far.

**Comment by janekste** [comment]
@LeeBamberTGC - I would like to have a minimum of 10 slots for audio files. Why is this necessary? I need this for universal dialog scripts. I use the maximum number of slots. I don't understand why this limit is needed. Why can't I control the number of sound slots for entities? Just add or remove the desired number of slots via lua? Any control option would have been fine with me. I want to control the number of sound fields for entities. However, 10 sound slots would be enough for me.

**Comment by tarkus1971** [comment]
At least 10 in second thoughts. Yes my initial idea was for dialog and in-game ambient effect that are entity specific. 15 or even more would be fantastic.

**Comment by AmenMoses** [comment]
If you really need more that 5 slots *per entity* then you can load sound files on the fly or use global sounds!

**Comment by tarkus1971** [comment]
would loading on the fly cause frame rate stutters?

**Comment by AmenMoses** [comment]
Not if you load at the right time, for example whilst the player is reading a note or between levels.
GG can hand-hold only up to a certain point, in fact I would argue that it tries to go too far as the excessive loading times people are always complaining about is almost all down to trying to make things easier for the user.
